theseus/core/tests/test_objective.py

```diff
@@ -280,6 +280,26 @@ def _check_error_for_data(v1_data_, v2_data_, error_, error_norm_2_):
         assert error.shape == (batch_size, 2 * dof)
         _check_error_for_data(v1_data, v2_data, error, error_norm_2)
 
+        # For input_data and also_update
+
+        v1_data = torch.ones(batch_size, dof) * f1 * 0.1
+        v2_data = torch.ones(batch_size, dof) * f2 * 0.1
+
+        input_data = {"v1": v1_data, "v2": v2_data}
+        error = objective.error(input_data=input_data, also_update=True)
+        error_norm_2 = objective.error_squared_norm(input_data=input_data, also_update=True)
+
+        _check_error_for_data(v1_data, v2_data, error, error_norm_2)
+
+        v1_data = torch.ones(batch_size, dof) * f1 * 0.4
+        v2_data = torch.ones(batch_size, dof) * f2 * 0.4
+
+        input_data = {"v1": v1_data, "v2": v2_data}
+        error = objective.error(input_data=input_data, also_update=True)
+        error_norm_2 = objective.error_squared_norm(input_data=input_data, also_update=True)
```

Comment:
We also need a test which covers the case where some input_data is passed and also_update is False.

Reply:
Agreed.